models/dvmdostem/R/write.config.dvmdostem.R

```diff
@@ -0,0 +1,353 @@
+  # to hold a representation of each dvmdostem parameter file.
+  dimveg_params <- paste(appbinary_path, "parameters", 'cmt_dimvegetation.txt', sep="/")
+  envcanopy_params <- paste(appbinary_path, "parameters", 'cmt_envcanopy.txt', sep="/")
+
+  dimveg_jsonfile <- '/tmp/dvmdostem-dimveg.json'
+  envcanopy_jsonfile <- '/tmp/dvmdostem-envcanopy.json'
+
+  # Call the helper script and write out the data to a temporary file
+  # This gets just the block we are interested in (based on community type)
+  system2(paste0(appbinary_path,"/scripts/param_util.py"),
+          args=(c("--dump-block-to-json", dimveg_params, cmtnum)),
+          stdout=dimveg_jsonfile, wait=TRUE)
+
+  system2(paste0(appbinary_path,"/scripts/param_util.py"),
+          args=(c("--dump-block-to-json", envcanopy_params, cmtnum)),
+          stdout=envcanopy_jsonfile, wait=TRUE)
+
+  # Read the json file into memory
+  library("rjson")
+  dimveg_jsondata <- fromJSON(paste(readLines(dimveg_jsonfile), collapse=""))
+  envcanopy_jsondata <- fromJSON(paste(readLines(envcanopy_jsonfile), collapse=""))
+
+  # (2)
+  # Overwrite certain parameter values with (ma-posterior) trait data
+  # from pecan, then write back out to disk...
+
+  # loop over all the incoming samples (traits)
+  for (cur_t_name in names(trait_df)) {
+    #print(cur_t_name)
+
+    # In converting trait.values into a dataframe, we get 
+    # these long column names that are dot separted and seem
+    # to contain cmtkey.pftname.traitname
+    # So the following splits out the various components into 
+    # convenient local variables
+    cmtkey <- unlist(strsplit(cur_t_name, '.', fixed=TRUE))[1]
+    pftname <- unlist(strsplit(cur_t_name, '.', fixed=TRUE))[2]
+    tname <- unlist(strsplit(cur_t_name, '.', fixed=TRUE))[3]
+
+    # NOTE:
+    # This is brittle with respect to changes in comments in dvmdostem
+    # parameter files. The parameter utility script determines names for
+    # variables based on parsing the comment string in the parameter files
+    # and looking for the part preceeding a colon. So changes in those
+    # files will require modifications to the following sections.
+
+    # Loop over all the items in our json structure of the dvmdostem parameters
+    # and find the item that matches the PFT
+    for (jd_name in names(dimveg_jsondata)) {
+      # First make sure that we are looking at the pftN entries, not 
+      # comments, or other non-PFT variables
+      if (grepl("pft", jd_name)) {
+        cur_pft_name <- dimveg_jsondata[[jd_name]]$name
+
+        # Then pick out only the entry for the correct pft 
+        if (identical(cur_pft_name, pftname)) {
+          print(cur_t_name)
+          print(tname)
+          if (tname == "SLA") {
+            dimveg_jsondata[[jd_name]]$sla = trait_df[[cur_t_name]]
+          } else if (tname == "frprod_perc_10") {
+            dimveg_jsondata[[jd_name]]$`frprod[0]` = trait_df[[cur_t_name]]
+          } else {
```

Comment:
1) The point of doing the earlier conversion from PEcAn names to Model names is to avoid having to so a big set of if's here -- if everything is in model names you should be able to just access the right variable directly
2) If you *do* end up with a big set of if's here (which isn't uncommon, I've done something similar in lots of models), then I really don't see the point of having a convert.samples function, since you can just do the name matching and unit conversions right here. Given what you've got here so far, it sure seems to me that the decision to write a convert.samples function has made things harder on you.